demo/src/main/java/com/novoda/storagepathfinder/demo/LandingActivity.kt

```diff
@@ -0,0 +1,44 @@
+package com.novoda.storagepathfinder.demo
+
+import android.os.Bundle
+import android.support.v7.app.AppCompatActivity
+import android.support.v7.widget.LinearLayoutManager
+import com.novoda.storagepathfinder.StoragePath
+import kotlinx.android.synthetic.main.activity_landing.*
+import java.util.*
+
+class LandingActivity : AppCompatActivity() {
+
+    private lateinit var assetCloner: AssetCloner
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_landing)
+
+        assetCloner = DemoDependenciesFactory.createAssetCloner(applicationContext)
```

Comment:
Application context is available before `onCreate` so it's safe to be instantiated and avoid `lateinit`. The other option would be to define it as `lazy`:
```
private val assetCloner by lazy { DemoDependenciesFactory.createAssetCloner(applicationContext) }
```
This means it's not a writable variable.